lib/frr_pthread.c

```diff
@@ -140,16 +140,25 @@ void frr_pthread_destroy(struct frr_pthread *fpt)
 	XFREE(MTYPE_FRR_PTHREAD, fpt);
 }
 
-void frr_pthread_set_name(struct frr_pthread *fpt, const char *name)
+int frr_pthread_set_name(struct frr_pthread *fpt, const char *name)
 {
+	int ret = 0;
+
 	pthread_mutex_lock(&fpt->mtx);
 	{
 		if (fpt->name)
 			XFREE(MTYPE_FRR_PTHREAD, fpt->name);
 		fpt->name = XSTRDUP(MTYPE_FRR_PTHREAD, name);
+#ifdef GNU_LINUX
+		ret = pthread_setname_np(fpt->thread, name);
```

Comment:
If this function, and the ones on other platforms, don't automatically truncate `name` to 15+1 characters we should do it here to be safe.